static/javascripts/components/survey_card.jsx

```diff
@@ -25,24 +25,36 @@ var SurveysPage = React.createClass({
 
     /*checks that all the required fields are filled*/
     checkSurvey: function(survey){
-        //console.log(survey);
+        //check group
+        if (survey.item_id ==""){
+            alert("Specify a group to post to");
+            return false;
+        }
         //check title
-        if (survey.item_name ===""){
+        if (survey.item_name ==""){
             alert("Complete the Survey Title");
             return false;
         }
         //check questions
         for (var i = 0; i< survey.questions.length; i++){
-            if(survey.questions[i].title === ""){
+            if(survey.questions[i].title == ""){
                 alert("Every Question should have a title.");
                 return false;
             }
-            if(survey.questions[i].response_format === "multipleChoice" || survey.questions[i].response_format ==="trueOrFalse"){
+            //two options minimum
+            if(survey.questions[i].response_format == "multipleChoice" || survey.questions[i].response_format =="trueOrFalse"){
                 if(survey.questions[i].options.length<2){
                 alert("Add at least two options.");
                 return false;
                 }
             }
+            //options must have titles
+            for(var i2 = 0; i2 < survey.questions[i].options.length; i2++){
+                if(survey.questions[i].options[i2].title == ""){
+                    alert("Options should have a title");
+                    return false;
+                }
+            }
```

Comment:
This is a for loop that runs on each question to make sure that the option fields aren't left empty so we don't have empty options available